src/simplewallet/simplewallet.cpp

```diff
@@ -4896,6 +4999,51 @@ boost::optional<epee::wipeable_string> simple_wallet::new_wallet(const boost::pr
   return password;
 }
 //----------------------------------------------------------------------------------------------------
+void simple_wallet::process_background_cache_on_open(const boost::program_options::variables_map& vm, const epee::wipeable_string& password)
```

Comment:
Note: I will likely end up not exposing this `process_background_cache_on_open` function from wallet2 so wallet API devs won't need to deal with it after implementing @kayabaNerve's suggestion here: https://github.com/monero-project/monero/pull/8619#issuecomment-1730458890

When loading a normal wallet, wallet2 can handle processing the background cache internally.